include/nccl_ofi_mr.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_MR_H_
+#define NCCL_OFI_MR_H_
+
+#ifdef _cplusplus
+extern "C" {
+#endif
+
+#include <pthread.h>
+
+/**
+ * A memory registration cache entry
+ */
+typedef struct nccl_ofi_reg_entry {
+	uintptr_t addr;
+	size_t pages;
+	int refcnt;
+	void *handle;
+} nccl_ofi_reg_entry_t;
+
+/**
+ * Device-specific memory registration cache.
+ */
+typedef struct nccl_ofi_mr_cache {
```

Comment:
Consider tracking hit/miss counters and printing that in the debug traces when tearing down this object.